db/snapshot/backup_task.cc

```diff
@@ -65,7 +65,7 @@ tasks::is_user_task backup_task_impl::is_user_task() const noexcept {
     return tasks::is_user_task::yes;
 }
 
-future<> backup_task_impl::upload_component(sstring name) {
+future<> backup_task_impl::upload_component(shared_ptr<s3::client>& client, sstring name) {
```

Comment:
Why pass the shared_ptr by reference? This is dangerous, if the backing store of the shared pointer is clobbered or disappears.

Also it makes it appear as an in/out parameter, which it is not.

Finally, it's slower, though that doesn't matter here.